hudi-client/hudi-client-common/src/test/java/org/apache/hudi/metadata/TestHoodieMetadataWriteUtils.java

```diff
@@ -41,7 +41,7 @@ public void testCreateMetadataWriteConfigForCleaner() {
         .build();
 
     HoodieWriteConfig metadataWriteConfig1 = HoodieMetadataWriteUtils.createMetadataWriteConfig(writeConfig1, HoodieFailedWritesCleaningPolicy.EAGER);
-    assertEquals(HoodieFailedWritesCleaningPolicy.EAGER, metadataWriteConfig1.getFailedWritesCleanPolicy());
+    assertEquals(HoodieFailedWritesCleaningPolicy.LAZY, metadataWriteConfig1.getFailedWritesCleanPolicy());
```

Comment:
In spark write client, we collect the `lastCompletedTxnAndMetadata` and `pendingInflightAndRequestedInstants` just for contlict resolution, we can eliminate that as an optimization under NB-CC, because NB-CC just skip the conflict resolution and these are useless.